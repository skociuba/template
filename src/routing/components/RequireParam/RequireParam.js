import React from 'react';
import {Navigate, useParams} from 'react-router-dom';
import PropTypes from 'prop-types';
import {getParsedChildItemFromStorage} from 'utils/hooks/useLocationState';

const LOCATION_STATE_KEY = 'locationState';

const RequireParam = ({children, isLocationStateRequired, redirectTo}) => {
  if (!isLocationStateRequired) {
    return children;
  }

  try {
    const {id} = useParams();

    const hasStoredParam = !!(
      getParsedChildItemFromStorage(localStorage, LOCATION_STATE_KEY, id) ||
      sessionStorage.getItem(id)
    );
    return hasStoredParam ? children : <Navigate to={redirectTo} />;
  } catch (error) {
    console.error('localStorage or sessionStorage is unavailable:', error);
  }
};

RequireParam.propTypes = {
  children: PropTypes.element.isRequired,
  isLocationStateRequired: PropTypes.bool.isRequired,
  redirectTo: PropTypes.string.isRequired,
};

export default RequireParam;
