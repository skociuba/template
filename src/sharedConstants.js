import {routes} from './routesConstants';

const shared = {
  routes: routes,
  header: {
    menu: {
      primary: [
        {name: 'Main', to: '/main-page'},
        {name: 'FrontendPaginationSorting', to: '/frontend-pagination-sorting'},
        {name: 'StaticTable', to: '/static-table-page'},
        {name: 'BackendControl', to: '/backend-control-page'},
        {name: 'FrontendControl', to: '/frontend-control-page'},
      ],
    },
  },
  mapping: [{value: 'testData', label: 'MappedData'}],
  statuses: {
    B: {
      title: 'BUY',
    },
    S: {
      title: 'SELL',
    },
  },
  names: {
    '603be08a2987d5267f590d2f': {
      title: '603be08a2987d5267f590d2f',
      isSelected: false,
    },
    '603be21b2987d50c34590d38': {
      title: '603be21b2987d50c34590d38',
      isSelected: false,
    },
    all: {
      title: 'all',
      isSelected: false,
    },
  },
};

export {shared};
