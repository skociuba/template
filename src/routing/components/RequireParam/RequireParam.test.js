import renderWithRouter from 'utils/tests/renderWithRouter';
import {shared} from 'sharedConstants';

import RequireParam from './RequireParam';
const paths = {...shared.routes};

const defaultProps = {
  children: <div>Page</div>,
  isLocationStateRequired: false,
  redirectTo: paths.frontendPaginationSorting.root,
};

describe('<RequireParam/>', () => {
  it('render without crashing', () => {
    const {container} = renderWithRouter(<RequireParam {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
