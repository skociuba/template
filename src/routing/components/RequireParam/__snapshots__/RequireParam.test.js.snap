// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequireParam/> render without crashing 1`] = `
<div>
  Page
</div>
`;
